pkg/unikernel/libvirt/vmtools.go

```diff
@@ -0,0 +1,388 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package libvirt
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	libvirtgo "github.com/libvirt/libvirt-go"
+	libvirtxml "github.com/libvirt/libvirt-go-xml"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/frakti/pkg/unikernel/metadata"
+)
+
+const (
+	DefaultMaxMem  = 32768 // size in MiB
+	DefaultMaxCPUs = 8
```

Comment:
why 8?

Reply:
No particular reason, the real limit is defined by kernel through CONFIG_NR_CPUS, this value should be configurable. And to reduce image size and memory cost, some simplified kernel(such as kernel shipped by hyperstart or clear containers) set this num as 8.